### DISCUSSION TITLE
Fosite: Client Registration flow

### DISCUSSION THREAD

**Discussion by RanjanKP45** [ory/.github#38] [Q&A]
Hi All,
I am looking into "**Client/App Registration**" into fosite, but couldn't able to find any doc./example for the same.
It would be great if someone help me out here

Thanks

**Comment by vinckr** [comment]
Hey @RanjanKP45,
Thanks for your interest in Ory Fosite!
Currently, we only have the [README](https://github.com/ory/fosite/blob/master/README.md) and a short guide for[ client credentials grant](https://github.com/ory/fosite/blob/master/docs/how-tos/client_credentials_grant.md).

You can also pose your question in the [community slack](https://slack.ory.sh/), "General" channel should also be good for Fosite.
I will keep my eyes open if I can find something that helps as well.

**Reply by RanjanKP45** [reply]
Hi @vinckr,
Thanks for the reply and sorry for coming back to you so late on this topic, I was side tracked for couple of months.
Currently I am using fosite.DefaultClient{} to register an client at the start. 
copied code snippet:
```
storage.Clients["test-client"] = &fosite.DefaultClient{
		ID:         "test-client",
		Secret:     []byte(`$2a$10$IxMdI6d.LIRZPpSfEwNoeu4rY3FhDREsxFJXikcgdRRAStxUlsuEO`), // = "foobar"
		GrantTypes: []string{"client_credentials"},
	}

```

What I was looking into "Registration of Client Dynamically and the Management", plan is to expose an endpoint where Admin/Super-User can access the endpoint to register the client dynamically. Something like this https://www.ory.sh/docs/hydra/reference/api#operation/createOAuth2Client in fosite. 

Thanks
